cloud/docker/docker.py

```diff
@@ -550,20 +551,23 @@
 
 
 def _human_to_bytes(number):
-    suffixes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB']
-
-    if isinstance(number, int):
-        return number
-    if number[-1] == suffixes[0] and number[-2].isdigit():
-        return number[:-1]
-
-    i = 1
-    for each in suffixes[1:]:
-        if number[-len(each):] == suffixes[i]:
-            return int(number[:-len(each)]) * (1024 ** i)
-        i = i + 1
-
-    raise ValueError('Could not convert %s to integer' % (number,))
+    suffixes = ['b','k','m','g','t','p']
+    # split input into tokens; if a dot is appeared in number then convert it to float
+    tokens = filter(lambda x: x != '',
+                    [float(x) if re.match(r'\d+\.\d+', x) else
+                     int(x) if x.isdigit() else x
+                     for x in map(lambda x: x.strip(), re.split(r'(\d+(?:\.\d+)?)', number))])
```

Comment:
This is a pretty complex line here: list comp + regex + inline conditionals + filter + map. Maybe it could be reworked as a loop with conditionals:

```
def coerce_token(token):
    token = token.strip()
    if re.match(r'\d+\.\d+', x):
        return float(token)
    elif x.isdigit():
        return int(token)
    else:
        return token

tokens = [coerce_token(t) for t in re.split(r'(\d+(?:\.\d+)?)', number) if t != '']
```